### ISSUE TITLE
OpenVPN speed problems

### ISSUE THREAD

**Issue by estimadarocha** [angristan/openvpn-install#31]
maybe this is not a issue...

i finish the installation of your script but i am getting very very slow speeds.

this is my setup:

home - 200mbit ubuntu desktop with i219-v ethernet

testing location - 100 mbits windows 10

i can't get more than 15 mbits.

i have pings of 17ms from one location to other without vpn... its really near

any idea @Angristan?

**Comment by superted999** [comment]
I didn't want to start a new thread but I have an issue with speed. 

Im using a Dedi Server from OVH running Ubuntu 16.10 Server.

Mine is slightly different from the one above. When I first connect I get the speeds I expect. Just a bit slower than using no VPN. However, over a period of time, say 24hr the speed reduces down significantly. 32mb down to 5mb. 

I then disconnect the VPN and re-connect and it goes back to 32mb. 

Does anyone know why?

**Comment by estimadarocha** [comment]
@superted999 can you give me one idea of chosen options in your setup?

**Comment by superted999** [comment]
I'll try and remember.......

On the original install went for all the reccomended/default settings. It did the same thing so when I reinstalled I chose all the options for the fatest to test this. Only thing I did change was the port number. Just tested again 4:35pm and still up at 32mb but I know when I check in the morning it will be slow again.

Tested again at 08:52 this morning and back down to 10mb. Disconnected, re-connected and back up to 32mb. Very odd @estimadarocha

**Comment by DrXala** [comment]
@estimadarocha @superted999 

Did you check the cpu load during your speed drop?
OpenVPN encrypts traffic which requires cpu. So if your cpu is loaded, your speed drops. And it's even more true if your encryption is strong.

I've already encountered this problem with some Internet provider routers that bore the encryption poorly.

**Comment by biffalo** [comment]
add the following lines to your openvpn config 
sndbuf `393216
rcvbuf 393216
push "sndbuf 393216"
push "rcvbuf 393216"
speed went from 20mb down to 67mb down after making that change.

**Comment by rhbkweb** [comment]
Im also having speed issues. Tryed to add the @biffalo lines to the opvn client file and it help a lot.

Before adding the lines I could not get more than 10Mbps and now Im able to get 20Mbps. But is still slow. 

I have a internet connection of 200Mbps and I tested my VPS download speed that is around 70Mbps. I tested both on Windows and Android phone.

VPN is installed on a new and clean Debian 9 64bit VPS server. Anyone has any advice or suggestion?

Thanks

**Comment by estimadarocha** [comment]
i also have the same problem!

**Comment by ipko1996** [comment]
tl;dr
**Try to buy vps service in your country.**

So, if anyone finds this, because seed problems I write my experience.
I started with
DigitalOcean, tried with Frankfurt, Debian, Ubuntu --> 5Mbps
okay the script can not be that wrong if just a few people have the same problem, 

> `sndbuf  393216`
> `rcvbuf 393216`
> `push "sndbuf 393216"`
> `push "rcvbuf 393216"`

did not work.

Linode, same setups, same result.
I guess I do something wrong, but I had an ovpn file from my friend and it worked just correctly. So I tracert-ed who hosts his vps service. It turns out none of the above 🙃
It was one of my country's vps providers.

So I was thinking what can I lose? About 3 dollars...
I registered, created the cheapest vps, set it up, and ran the script.

Guess what! It was perfect. No speed issues or ping problems.
So my suggestion is that if anyone have these problems and modifying the openvpn's config file does not work,
you should try to buy a vps in your country if you are able to do that.
